### PR TITLE
feat: delete confirmation dialogs [BD-38] [TNL-9351]

### PR DESCRIPTION
**Pull request by xitij2000**
Adds delete confirmation dialogs for posts, comments and replies.

Screenshot_20220118_181702

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @xitij2000! I've created [OSPR-6402](https://openedx.atlassian.net/browse/OSPR-6402) to keep track of it in JIRA, where we prioritize reviews. Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

File: src/discussions/comments/comment/Reply.jsx [hunk 1]
```diff
@@ -5,14 +5,16 @@ import { useDispatch, useSelector } from 'react-redux';
...
 
 import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
-import { Avatar } from '@edx/paragon';
+import { Avatar, useToggle } from '@edx/paragon';
 
 import { ContentActions } from '../../../data/constants';
+import { DeleteConfirmation } from '../../common';
 import ActionsDropdown from '../../common/ActionsDropdown';
 import AlertBanner from '../../common/AlertBanner';
 import AuthorLabel from '../../common/AuthorLabel';
```
Comment (kaustavb12) [inline src/discussions/comments/comment/Reply.jsx:14]: Nit:

```suggestion
import { ActionsDropdown, AlertBanner, AuthorLabel, DeleteConfirmation } from '../../common';
```

File: src/discussions/posts/post/Post.jsx [hunk 2]
```diff
@@ -27,16 +27,30 @@ function Post({
...
-    <div className="d-flex flex-column p-2.5 w-100 mw-100">
+    <div className="d-flex flex-column p-2.5 w-100 mw-100" data-testid={`post-${post.id}`}>
+      <DeleteConfirmation
+        isOpen={isDeleting}
+        title={intl.formatMessage(messages.deletePostTitle)}
+        description={intl.formatMessage(messages.deletePostDescription)}
+        onClose={hideDeleteConfirmation}
+        onDelete={() => {
+          dispatch(removeThread(post.id));
+          history.goBack();
```
Comment (kaustavb12) [inline src/discussions/posts/post/Post.jsx:50]: If we use `history.goBack()`, then there is a discrepency in the user experience based on if the deleted post is the first one the user had opened or if the user comes to this post from some other post.

To provide a uniform experience to the user, I would suggest using this :
```suggestion
          history.push(".");
```

**Review by kaustavb12 — APPROVED** [review]
:+1: 
I have left two comments.
All other changes looks good to go.

This PR is approved on the condition of resolution of the comments.

- [x] I tested this: tested the delete confirmation dialog in devstack
- [x] I read through the code

File: src/discussions/posts/post/Post.jsx (hunk 2 above)
Comment (xitij2000) [reply src/discussions/posts/post/Post.jsx]: I didn't know about this! Thanks!

**Review by mehaknasir — APPROVED** [review]

**Review by awais-ansari — APPROVED** [review]

**Comment by edx-webhook** [comment]
@xitij2000 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-discussions&entry.867055334=https://github.com/openedx/frontend-app-discussions/pull/49&entry.1484655318=https://github.com/xitij2000&entry.752974735=2022-01-18+12:46&entry.1917517419=2022-01-27+07:18&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**